wa/workloads/exoplayer/__init__.py

```diff
@@ -0,0 +1,185 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2017, Arm Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from collections import defaultdict
+import re
+import os
+import time
+import urllib
+
+from wa import ApkWorkload, Parameter, ConfigError, WorkloadError
+from wa.framework.configuration.core import settings
+from wa.utils.types import boolean
+from wa.utils.misc import ensure_directory_exists
+from devlib.utils.android import grant_app_permissions
+
+# Regexps for benchmark synchronization
+REGEXPS = {
+    'start'    : '.*Displayed com.google.android.exoplayer2.demo/.PlayerActivity',
+    'duration' : '.*period \[(?P<duration>[0-9]+.*)\]',
+    'end'      : '.*state \[.+, .+, E\]'
+}
+
+
+DOWNLOAD_URLS = {
+    'mp4_1080p': 'http://distribution.bbb3d.renderfarming.net/video/mp4/bbb_sunflower_1080p_30fps_normal.mp4',
+    'mov_720p':  'http://download.blender.org/peach/bigbuckbunny_movies/big_buck_bunny_720p_h264.mov',
+    'mov_480p':  'http://download.blender.org/peach/bigbuckbunny_movies/big_buck_bunny_480p_h264.mov',
+}
+
+
+class ExoPlayer(ApkWorkload):
+    """
+    Android ExoPlayer (YouTube)
+
+    ExoPlayer is the basic video player library that is used by the YouTube
+    android app. The aim of this workload is to test a proxy for YouTube
+    performance on targets where running the real YouTube app is not possible
+    due its dependencies.
+
+    ExoPlayer sources: https://github.com/google/ExoPlayer
+
+    The 'demo' application is used by this workload.  It can easily be built by
+    loading the ExoPlayer sources into Android Studio.
+
+    Version r2.4.0 built from commit d979469 is known to work
+    """
+
+    name = 'exoplayer'
+
+    video_directory = os.path.join(settings.dependencies_directory, name)
+
+    versions = ["2.4.0"]
+    action = 'com.google.android.exoplayer.demo.action.VIEW'
+
+    parameters = [
+        Parameter('package_name', default='com.google.android.exoplayer2.demo', override=True),
```

Comment:
This should be set as `package_names` class attribute rather than overriding the `package_name` parameter.